pkg/controller/remotemachineset/leaseexceptions.go

```diff
@@ -40,6 +44,47 @@ func (h *machinePoolNameLeaseEventHandler) Create(e event.CreateEvent, q workque
 	h.EnqueueRequestForOwner.Create(e, q)
 }
 
+// Delete implements handler.EventHandler
+func (h *machinePoolNameLeaseEventHandler) Delete(e event.DeleteEvent, q workqueue.RateLimitingInterface) {
+	logger := h.reconciler.logger
+	logger.Info("running Delete handler for MachinePoolNameLease, requeuing all pools for cluster")
+	lease, ok := e.Object.(*hivev1.MachinePoolNameLease)
+	if !ok {
+		logger.Warn("Delete handler called for non-MachinePoolNameLease: %v", e.Object)
+		return
+	}
+	if _, ok := lease.Labels[constants.ClusterDeploymentNameLabel]; !ok {
+		logger.WithFields(log.Fields{
+			"lease":     lease.Name,
+			"namespace": lease.Namespace,
+		}).Warnf("deleted lease has no %s label, unable to requeue all pools for cluster", constants.ClusterDeploymentNameLabel)
+		return
+	}
+	logger.Info("listing all pools for cluster deployment")
+
+	// If a lease is deleted, requeue all the pools for the cluster, somebody might be waiting for a lease to free up.
+	// TODO: we do not currently label machine pools as belonging to the cluster, when we do we could use this to filter
+	// during the query itself.
+	clusterMachinePools := &hivev1.MachinePoolList{}
+	err := h.reconciler.List(context.TODO(), clusterMachinePools, client.InNamespace(lease.Namespace))
+	if err != nil {
+		logger.WithError(err).Log(controllerutils.LogLevel(err),
+			"unable to list machine pools for cluster")
+		return
+	}
+	logger.Debugf("found %d MachinePools for cluster", len(clusterMachinePools.Items))
+	for _, mp := range clusterMachinePools.Items {
+		if mp.Spec.ClusterDeploymentRef.Name != lease.Labels[constants.ClusterDeploymentNameLabel] {
+			continue
+		}
+		h.EnqueueRequestForOwner.Delete(e, q)
```

Comment:
Remove this line. This will requeue the owner of the deleted machinepoolnamelease.